File: toy/sns/src/main/java/me/kzv/sns/controller/UserController.java
package me.kzv.sns.controller;

import lombok.RequiredArgsConstructor;
import me.kzv.sns.controller.request.UserJoinRequest;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/v1/users")
public class UserController {

    private final UserService userService;

    @PostMapping("/join")
    public void join(@RequestBody UserJoinRequest request){
        userService.join(request.getUserName(), request.getPassword());
    }

}

File: toy/simple-sns/src/main/kotlin/me/kzv/simplesns/controller/UserController.kt
package me.kzv.simplesns.controller

import me.kzv.simplesns.controller.request.UserJoinRequest
import me.kzv.simplesns.controller.response.Response
import me.kzv.simplesns.controller.response.UserJoinResponse
import me.kzv.simplesns.service.UserService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/users")
class UserController(
    private val userService: UserService,
) {
    @PostMapping("/join")
    fun join(@RequestBody request: UserJoinRequest): Response<UserJoinResponse> {
        val user = userService.join(userName = request.userName, password = request.password)
        return Response.success(UserJoinResponse.fromUser(user))
    }
}